Maintenance of VPC module

tpdownes opened the pull request:
- update to CFT Cloud Router module 2.0
- add support for custom firewall rules
- optionally enable/disable default firewall rules
- add support for secondary IP ranges in VPC as "passthrough" variable
- update documentation to direct users for information on detailed structure of the complex object/map variables

I have manually tested the `hpc-small.yaml` example and observed that the default firewall rules are still created. I have also manually submitted integration group 4 build and observed success. I believe only groups 3 and 4 test this module's functionality.

### Submission Checklist

* [X] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [X] Are all tests passing? (`make tests`)
* [X] Have you written unit tests to cover this change?
* [X] Is unit test coverage still above 80%?
* [X] Have you updated all applicable documentation?
* [X] Have you followed the guidelines in our Contributing document?

nick-stroud commented on modules/network/vpc/main.tf:
```diff
@@ -65,64 +117,13 @@ module "vpc" {
   delete_default_internet_gateway_routes = var.delete_default_internet_gateway_routes
-}
-
-module "firewall_rules" {
```
What is the reason behind no longer using this module? Advantages & disadvantages?

tpdownes replied:
I have chosen to revert the migration because it doesn't cleanly handle create/destroy for existing infra. We should wait until we are willing to impose Terraform 1.1 (or Terraform can silently ignore blocks it doesn't yet understand) to do this

nick-stroud commented on modules/network/vpc/README.md:
```diff
@@ -1,20 +1,28 @@
-[Cloud Foundation Toolkit](https://cloud.google.com/foundation-toolkit).
-
-The created cloud NAT (Network Address Translation) allows virtual machines
-without external IP addresses to create outbound connections to the internet.
```
nit: "without external IP addresses" was informative (at least to me) and was  not carried over. Considering adding it to the updated version.

tpdownes replied:
Addressed nit on HEAD of PR

nick-stroud commented on modules/network/vpc/README.md:
```diff
@@ -1,20 +1,28 @@
-* Allow SSH access from the Cloud Console ("35.235.240.0/20").
-* Allow traffic between nodes within the VPC
+This module creates a new [VPC network][vpc], subnetworks, and [Cloud
+Router][router]. By default, it will enable
```
nit: This list reads odd to me, I think because you use the verb `enable` to introduce the list and then all items also use `enable`. Maybe if you s/it will enable/it will create/.

nick-stroud commented on modules/network/vpc/README.md:
```diff
@@ -1,20 +1,28 @@
+This behavior is optional and can be configured as [described below](#inputs).
+This module is based on networking support in the [Cloud Foundation
+Toolkit][cft]. We recommend following the [documentation for the network
+module][cft-network] and [submodules][cft-network-submodules] for detailed
```
nit: The links here take a little poking around to find what you are referencing. 

The `cft-network` link has the `secondary_ranges` definition. To get the `firewall_rules` definition I go to `cft-network-submodules` link and then click on `firewall-rules`.  

Why not just link directly to the firewall-rules module?
Why not put the link next to the respective variable so it is clear what to click on for what?

tpdownes replied:
Addressed nit on HEAD of PR with a bulleted list toward top. I think we need a broader strategy for `variables.tf` documentation that doesn't involve clunky multi-line HEREDOCs with extremely long URLs.

tpdownes commented on modules/network/vpc/README.md:
```diff
@@ -1,20 +1,28 @@
-
-* Allow SSH access from the Cloud Console ("35.235.240.0/20").
-* Allow traffic between nodes within the VPC
+This module creates a new [VPC network][vpc], subnetworks, and [Cloud
```
The accurate text ought to be "a Cloud Router in every region with a subnetwork". I'll massage this a bit.

tpdownes commented:
One of your comments prompted a real change to keep the distinct `firewall_rules` module until we adopt Terraform 1.1+. While doing this, addressed the nits. The real change required re-writing the git history so there is a force push here. Hopefully conversations not too impacted.
